test/metabase/upload_test.clj

```diff
@@ -506,16 +514,17 @@
 (deftest load-from-csv-table-name-test
   (testing "Can upload two files with the same name"
     (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
-      (mt/with-empty-db
```

Comment:
The main culprit is `mt/with-empty-db`, which doesn't do anything for redshift but sync the test database.

Reply:
We should probably put a warning in the `mt/with-empty-db` docstring about this footgun for redshift, and give the guidance to use something table-level.

Reply:
6e0cabc9ecce5cfc82255e82c484959eb8426e70

Reply:
New docstring: 
"Sets the current dataset to a freshly created db that gets destroyed at the conclusion of `body`.
   Use this to test destructive actions that may modify the data.
   WARNING: this doesn't actually create and destroy a temporary database for cloud databases (like redshift) that
   reuse a single database for all tests."